packages/jobs/lib/execution/sync.ts

```diff
@@ -295,6 +300,34 @@ export async function handleSyncSuccess({ nangoProps }: { nangoProps: NangoProps
                 });
                 void tracer.scope().activate(span, async () => {
                     try {
+                        if (!team) {
+                            throw new Error('Missing team');
+                        }
+
+                        if (!providerConfig.id) {
+                            throw new Error('Missing providerConfig.id');
+                        }
+
+                        if (!connection.id) {
+                            throw new Error('Missing connection.id');
+                        }
+
+                        if (!nangoProps.syncConfig.id) {
```

Comment:
These checks are how I ended up resolving the build issue in CI. More than happy to discuss further, but basically the lint error I was seeing in the build history on commits in this PR wasn't happening on my computer, and I could never get it to reproduce. This worked around it.

Reply:
I think you are just not up to date